### ISSUE TITLE
Support for super users

### ISSUE THREAD

**Issue by numberten** [numberten/zhenya_bot#45]
I think we should add some formal support for 'super users' or higher tiered users. This would allow for things like #44. Any ideas for a good way of doing this? 

I was thinking a nice way of doing this would be passing a list of approved nicks as command line args and storing them as bot state accessible to any component. I'm unsure how nick clusters would be treated though. Not respecting clusters would be annoying, but respecting them could be a security hole.

**Comment by fimad** [comment]
I don't think we should obey nick clusters for super users. Too easy to game.

I think the most extensible way to do this would be like the list component. Super user nicks are stored in a data file and there is a privileged command to add and remove users from that list.

**Comment by numberten** [comment]
:+1:

**Comment by numberten** [comment]
How do the first super users get authenticated? Manually editing the data/superusers file before your first run is kinda janky.

**Comment by fimad** [comment]
Couple of ideas, not sure if any are good.
1. Automatically add a nick to the file if they are op and they grant ops to the bot.
2. When joining a channel, automatically add all ops to the list of super users for that channel.
3. Set a password as a command line flag. Anyone who pm's zbot the password gets added to the list of super users.

**Comment by numberten** [comment]
The first two would work well on a ChanServ registered channel (so on freenode) where you don't automatically become an op by being the first one to connect in a channel. Probably not so good for other ircds or unregistered channels.

I really like 3 though, I think that would work well.
